sway-core/src/semantic_analysis/program.rs

```diff
@@ -196,6 +196,29 @@ impl TypedProgram {
         ok(typed_program_kind, vec![], errors)
     }
 
+    /// Checks that no arguments passed to a `main()` in a `script` or `predicate`.
+    pub(crate) fn check_args_to_main(&self) -> CompileResult<()> {
+        let mut errors = Vec::new();
+        let warnings = Vec::new();
+        match &self.kind {
+            TypedProgramKind::Script{main_function,..} |
+            TypedProgramKind::Predicate{main_function, ..}  => {
+                if !main_function.parameters.is_empty() {
+                    errors.push(CompileError::MainArgsNotYetSupported {
+                        span: main_function.span.clone()
+                    })
+                }
+            },
+            _ => (),
+        }
+
+        CompileResult{
+            value:None,
+            warnings,
+            errors,
+        }
```

Comment:
Rather than create a `CompileResult` directly you should use `ok()` or `err()`, particularly since this isn't quite valid.  A `CompileResult` with a `None` value usually indicates an error.  I'll make an issue for this.
```suggestion
        ok((), warnings, errors)
```